src/psyclone/psyir/transformations/omp_task_trans.py

```diff
@@ -0,0 +1,158 @@
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author A. B. G. Chalk STFC Daresbury Lab
+''' This module provides the OMPTaskTrans transformation.'''
+from psyclone.errors import GenerationError
+from psyclone.domain.common.transformations import KernelModuleInlineTrans
+from psyclone.psyGen import Kern
+from psyclone.psyir.transformations.fold_conditional_return_expressions_trans \
+        import FoldConditionalReturnExpressionsTrans
+from psyclone.psyir.transformations.parallel_loop_trans import\
+    ParallelLoopTrans
+from psyclone.psyir.nodes import CodeBlock, Call
+from psyclone.psyir.nodes import DynamicOMPTaskDirective
+from psyclone.psyir.transformations.inline_trans import InlineTrans
+from psyclone.psyir.transformations.transformation_error import \
+        TransformationError
+
+
+class OMPTaskTrans(ParallelLoopTrans):
+    ''' Apply an OpenMP Task Transformation to a Loop. The Loop must
+    be within an OpenMP Serial region (Single or Master) at codegen time.
+    '''
+
+    def __str__(self):
+        return "Adds an 'OMP TASK' directive to a statement"
+
+    @property
+    def name(self):
+        '''
+        :returns: the name of this transformation.
+        :rtype: str
+        '''
+        return "OMPTaskTrans"
+
+    def validate(self, node, options=None):
+        '''
+        Validity checks for input arguments.
+
+        :param node: the Loop node to validate.
+        :type node: :py:class:`psyclone.psyir.nodes.Loop`
+        :param options: a dictionary with options for transformations.
+        :type options: dict of string:values or None
+        '''
+        # Disallow CodeBlocks inside the region
+        if len(node.walk(CodeBlock)) > 0:
```

Comment:
As @hiker said before, I think `if any(node.walk(CodeBlock)):` could potentially be more performant.

Reply:
Changed.